### PR TITLE
#712 (Backup & Restore) updated/cleaned up

### PR DESCRIPTION
**Pull request by erikdesjardins**
Closes #712.

Someone should test this in Safari.

### CODE REVIEW

File: lib/modules/backupAndRestore.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+					return;
+				}
+				// Reimport data
+				for (var k in settings) {
+					RESStorage.setItem(k, settings[k]);
+				}
+
+				alert('Your RES settings have been imported. Reloading reddit.');
+				location.hash = '';
+				setTimeout(location.reload.bind(location), 500);
```
Comment (mc10) [inline lib/modules/backupAndRestore.js]: Is the `bind` here necessary?

Comment (erikdesjardins) [inline lib/modules/backupAndRestore.js]: It is, apparently:
image

File: lib/modules/backupAndRestore.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		}
+	});
+
+	function backup() {
+		// Generate copy of RES settings for download
+		var settings = JSON.stringify(RESStorage);
+		var link = document.createElement('a');
+		if (BrowserDetect.isChrome()) {
+			// Encoding can make the URI long enough to crash Chrome; it doesn't appear to care if it's unencoded
+			link.href = 'data:text/plain;charset=utf-8,' + settings;
```
Comment (JordanMilne) [inline lib/modules/backupAndRestore.js]: You can get around this by [storing the json in a `Blob`](http://jsfiddle.net/koldev/cw7w5/) and setting `.href` to the `Blob`'s URI instead.

Comment (erikdesjardins) [inline lib/modules/backupAndRestore.js]: Ooooh, that's perfect.

**Pull request merged**